test/e2e/network/endpointslice.go

```diff
@@ -303,6 +304,131 @@ var _ = SIGDescribe("EndpointSlice", func() {
 		expectEndpointsAndSlices(cs, f.Namespace.Name, svc2, []*v1.Pod{pod1, pod2}, 2, 2, true)
 	})
+
+	ginkgo.It("should NOT include terminating endpoints for Endpoints and EndpointSlices", func() {
+		labelPod1 := "pod1"
+		labelValue := "on"
+
+		pod1 := podClient.Create(&v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "pod1",
+				Labels: map[string]string{
+					labelPod1: labelValue,
+				},
+			},
+			Spec: v1.PodSpec{
+				// use long grace periods to leave time to test endpoints
+				TerminationGracePeriodSeconds: utilpointer.Int64Ptr(120),
+				Containers: []v1.Container{
+					{
+						Name:    "container1",
+						Image:   imageutils.GetE2EImage(imageutils.BusyBox),
+						Command: []string{"sleep", "1000000"},
+						Ports: []v1.ContainerPort{{
+							Name:          "example-name",
+							ContainerPort: int32(3000),
+						}},
+					},
+				},
+			},
+		})
+
+		svc1 := createServiceReportErr(cs, f.Namespace.Name, &v1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "example-int-port",
+			},
+			Spec: v1.ServiceSpec{
+				Selector: map[string]string{labelPod1: labelValue},
+				Ports: []v1.ServicePort{{
+					Name:       "example",
+					Port:       80,
+					TargetPort: intstr.FromInt(3000),
+					Protocol:   v1.ProtocolTCP,
+				}},
+			},
+		})
+
+		// wait for pod to start up
+		err := wait.Poll(5*time.Second, time.Minute, func() (bool, error) {
+			var err error
+			pod1, err = podClient.Get(context.TODO(), pod1.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			if len(pod1.Status.PodIPs) == 0 {
+				return false, nil
+			}
+
+			return true, nil
+		})
+		framework.ExpectNoError(err, "timed out waiting for Pods to have IPs assigned")
+
+		ginkgo.By("referencing a single matching pod")
+		expectEndpointsAndSlices(cs, f.Namespace.Name, svc1, []*v1.Pod{pod1}, 1, 1, false)
+
+		// Delete pod1 with long termination grace period.
+		// NOTE: for this test we need to use a pod that ignores SIGTERM from kubelet
+		// in order to keep terminating pods around long enough to test this behavior.
+		err = podClient.Delete(context.TODO(), pod1.Name, metav1.DeleteOptions{})
+		framework.ExpectNoError(err, fmt.Sprintf("failed to delete pod %s/%s", pod1.Namespace, pod1.Name))
+
+		// ensure deletion timestamp
+		err = wait.Poll(5*time.Second, time.Minute, func() (bool, error) {
+			var err error
+			pod1, err = podClient.Get(context.TODO(), pod1.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+
+			if pod1.DeletionTimestamp == nil {
+				return false, nil
+			}
+
+			return true, nil
+		})
+		framework.ExpectNoError(err, "timed out waiting for pod to have deletion timestamp")
+
+		// TODO: update this test to include terminating endpoints once EndpointSliceTerminatingCondition is enabled by default
+		ginkgo.By("checking terminating endpoints are NOT included in EndpointSlice")
+		err = wait.Poll(2*time.Second, 10*time.Second, func() (bool, error) {
```

Comment:
Verify with @robscott the timers, I remember some slices e2e had to bump them because were operating with short timeouts that creating flakines

Reply:
I think those previous flakes were related to EndpointSlice garbage collection when Services were deleted. That scenario likely needed longer poll times since garbage collection happened every 30 seconds. 

This is testing that endpoints are immediately removed from the slice when pods are deleted. I think a shorter and stricter timeout like 10s should be sufficient. We should actually be a little concerned if this test started to flake.